code/modules/heavy_vehicle/mecha.dm

```diff
@@ -206,6 +206,8 @@
 		if(source_frame.body)
 			source_frame.body.forceMove(src)
 			body = source_frame.body
+			if(body.cell)
+				RegisterSignal(body.cell, COMSIG_CELL_CHARGE, PROC_REF(handle_cell_charge))
```

Comment:
Unregister it on destroy

Reply:
as far as i'm aware, signals are already unregistered when any datum gets destroyed, so it shouldn't be necessary, unless i missed something?

Reply:
You might be right